Azure/sonic-buildimage
# Dell S6000 ON related Port numbering and eeprom changes

**What i did**
Modified the front-end port to hardware port lane mapping, broadcom logical port to hardware port lane mapping for Dell S6000 ON.
Modified eeprom.py for S6000 ON. The lane mapping and eeprom.py currently have been based out of non ON S6000 variant, so if anyone uses sonic image on Dell S6000 ON variant will notice a difference in numbering between the front-panel and command line interface.

**How I did** 
Modified port_map.ini to swap the hardware to interface name so that interface name maps to front panel numbering.
Modified td2-s6000-32x40G.config.bcm so that broadcom logical port to physical port numbering is in similar fashion 
Modified eeprom.py to extract eeprom information in standard format rather than non ON S6000 format.

**How to verify it** 
Test show lldp table output before and after the change. Below is the change in behavior. Also test by connecting multiple ports back to back, connect with a peer device, etc. Check if the port state changes are reflected correctly in 'show interface status'

Before the change:
root@sonic:~# show lldp table
Capability codes: (R) Router, (B) Bridge, (O) Other
LocalPort    RemoteDevice    RemotePortID    Capability    RemotePortDescr
-----------  --------------  --------------  ------------  -----------------
Ethernet32   sonic           fortyGigE0/100  BR            Ethernet100
Ethernet84   sonic           fortyGigE0/88   BR            Ethernet88
Ethernet88   sonic           fortyGigE0/84   BR            Ethernet84
Ethernet100  sonic           fortyGigE0/32   BR            Ethernet32
--------------------------------------------------
Total entries displayed:  4
root@sonic:~#

After the change:
root@sonic:~# show lldp table
Capability codes: (R) Router, (B) Bridge, (O) Other
LocalPort    RemoteDevice    RemotePortID    Capability    RemotePortDescr
-----------  --------------  --------------  ------------  -----------------
Ethernet36   sonic           fortyGigE0/96   BR            Ethernet96
Ethernet80   sonic           fortyGigE0/92   BR            Ethernet92
Ethernet92   sonic           fortyGigE0/80   BR            Ethernet80
Ethernet96   sonic           fortyGigE0/36   BR            Ethernet36
--------------------------------------------------
Total entries displayed:  4

For eeprom.py change test is done using show platform syseeprom
**Before the change:**
root@sonic:~# show platform syseeprom
TLV Name             Len Value
-------------------- --- -----
Invalid TLV field starting at EEPROM offset 0
(checksum valid)
root@sonic:~#

**After the change:**
root@sonic:~# show platform syseeprom
TlvInfo Header:
   Id String:    TlvInfo
   Version:      1
   Total Length: 80
TLV Name             Code Len Value
-------------------- ---- --- -----
Product Name         0x21   8 S6000-ON
Base MAC Address     0x24   6 90:B1:1C:F4:9C:21
Vendor Name          0x2D   7 DELLEMC
Serial Number        0x23   7 6V71VS1
Platform Name        0x28  26 x86_64-dell_s6000_s1220-r0
ONIE Version         0x29   8 3.20.1.4
CRC-32               0xFE   4 0xDF2D2EFE
(checksum valid)
root@sonic:~#

For broadcom port mapping testing is done using bcmcmd "ps"
